src/components/common/Navbar/Navbar.tsx

```diff
@@ -42,30 +42,51 @@ function NavLinks({ onClickLink }: NavLinksProps) {
 
 function Navbar() {
   const [showMenu, setShowMenu] = useState(false)
+  const [searchExpanded, setSearchExpanded] = useState(false)
   const dismissTransition = useRef<Callback | undefined>()
   const handleCloseSlideOver = () => setShowMenu(false)
 
   return (
     <header className="navbar / grid-content-full">
       <div className="navbar__header / grid-content">
         <section className="navbar__row">
-          <IconButton
-            classes="navbar__menu"
-            aria-label="Open Menu"
-            icon={<ListIcon size={32} />}
-            onClick={() => setShowMenu(true)}
-          />
-          <LinkGatsby
-            to="/"
-            aria-label="Go to Faststore home"
-            title="Go to Faststore home"
-            className="navbar__logo"
-          >
-            <Logo />
-          </LinkGatsby>
+          {!searchExpanded && (
+            <>
+              <IconButton
+                classes="navbar__menu"
+                aria-label="Open Menu"
+                icon={<ListIcon size={32} />}
+                onClick={() => setShowMenu(true)}
+              />
+              <LinkGatsby
+                to="/"
+                aria-label="Go to Faststore home"
+                title="Go to Faststore home"
+                className="navbar__logo"
+              >
+                <Logo />
+              </LinkGatsby>
+            </>
+          )}
           <SearchInput />
-          <div className="navbar__buttons">
-            <SignInLink />
+          <div
+            className="navbar__buttons"
+            data-store-search-expanded={searchExpanded}
+          >
+            {searchExpanded && (
+              <IconButton
+                classes="retract__button"
+                aria-label="Retract Searchbar"
+                icon={<CaretLeft size={32} />}
+                onClick={() => setSearchExpanded(false)}
+              />
+            )}
+            <SearchInput
+              placeholder=""
+              testId="store-input-mobile"
+              buttonTestId="store-input-mobile-button"
+              onMagnifierClick={() => setSearchExpanded(true)}
+            />
```

Comment:
One thing I didn't understand. Why are you rendering the `<SearchInput>` twice? (in lines 71 and 84)

Reply:
Because the search input mobile and desktop have very different behaviours, because of this I preferred to have two search bars and hide depending on the device

Reply:
Yes, is possible for the input focus, but I will need to change some things in Search Bar in faststore, i going to try to do this